run_wml_tests

```diff
@@ -155,70 +182,71 @@ class WesnothRunner:
             args.append("-u")
             args.append(test.name)
+
         if self.timeout == 0:
-            if test.status == UnitTestResult.TIMEOUT:
-                test_summary.skip_test()
-                print('Skipping test', test_list[0].name, 'because timeout is disabled')
-                return
             timeout = None
+        elif len(test_list) == 1:
+            timeout = self.timeout
         else:
-            if test.status == UnitTestResult.FAIL_BROKE_STRICT and not options.strict_mode:
-                test_summary.skip_test()
-                print('Skipping test', test_list[0].name, 'because strict mode is disabled')
-                return
-            if len(test_list) == 1:
-                timeout = self.timeout
-            else:
-                timeout = self.batch_timeout
-        if self.verbose > 0:
+            timeout = self.batch_timeout
+
+        if self.verbose >= Verbosity.NAMES_OF_TESTS_RUN:
             if len(test_list) == 1:
                 print("Running test", test_list[0].name)
             else:
                 print("Running {count} tests ({names})".format(count=len(test_list),
                     names=", ".join([test.name for test in test_list])))
-            if self.verbose > 1:
-                print(repr(args))
+        if self.verbose >= Verbosity.SCRIPT_DEBUGGING:
+            print(repr(args))
+
         try:
             res = run_with_rerun_for_sdl_video(args, timeout)
         except subprocess.TimeoutExpired as t:
-            if self.verbose > 0:
-                print("Timed out (killed by Python timeout implementation)")
+            if expected_result != UnitTestResult.TIMEOUT:
+                print("Timed out (killed by Python timeout implementation), while running batch ({names})".format(
+                    names=", ".join([test.name for test in test_list])))
             res = subprocess.CompletedProcess(args, UnitTestResult.TIMEOUT.value, t.output or b'')
-        if self.verbose > 0:
+        if self.verbose >= Verbosity.OUTPUT_OF_PASSING_TESTS:
             print(res.stdout.decode('utf-8'))
-            if self.verbose > 1:
-                print("Result:", res.returncode)
-        num_passed = 1
-        if test_list[0].status == UnitTestResult.PASS:
-            num_passed = res.stdout.count(b"PASS TEST")
-            test_summary.pass_test(num_passed)
+            print("Result:", res.returncode)
         if res.returncode < 0:
             print("Wesnoth exited because of signal", -res.returncode)
             if options.backtrace:
                 print("Launching GDB for a backtrace...")
                 gdb_args = ["gdb", "-q", "-batch", "-ex", "start", "-ex", "continue", "-ex", "bt", "-ex", "quit", "--args"]
                 gdb_args.extend(args)
                 subprocess.run(gdb_args, timeout=240)
-            test_summary.crash_test()
-            test_summary.skip_test(len(test_list) - num_passed - 1)
+            test_summary.crash_test(test_list)
```

Comment:
Can you still see which test in the batch actually crashed from this?

Reply:
No it doesn't work out which one of the batch was running when the crash happened. However, given it could be memory corruption from an earlier test and that it will try to replicate the test and generate a stacktrace, I'm not sure that it could give much more info. If we get a stacktrace then that's going to be much better and more reliable info, and if we don't get a stacktrace then it's probably better to expect the developer to replicate it themselves rather than trying to script logic about guessing what was running based on what output had been printed by Wesnoth.

Reply:
I think it's useful to show what test was running but you make a good point about the possibility of the crash being caused by an earlier test. A stack trace can also help of course, but the name of the test should at least give a starting point in investigating where the crash occurred.

On a side note, it'd be great if we could get the stack trace even on Windows… also, the gdb stack trace functionality should probably be enabled also on Mac (though maybe using lldb instead).

Reply:
That seems to be a wishlist for future additions; are there any changes needed before merging the current PR?

Reply:
Yeah, that's pretty much accurate. There are no changes needed before merging here.